scripts/generateChangelog.js

```diff
@@ -0,0 +1,158 @@
+/* eslint-disable no-restricted-syntax */
+const { Octokit } = require('@octokit/rest');
+const yargs = require('yargs');
+
+/**
+ * @param {string} commitMessage
+ * @returns {string} The tags in lowercases, ordered ascending and commaseparated
+ */
+function parseTags(commitMessage) {
+  const tagMatch = commitMessage.match(/^(\[[\w-]+\])+/);
+  if (tagMatch === null) {
+    return '';
+  }
+  const [tagsWithBracketDelimiter] = tagMatch;
+  return tagsWithBracketDelimiter
+    .match(/([\w-]+)/g)
+    .map((tag) => {
+      return tag.toLocaleLowerCase();
+    })
+    .sort((a, b) => {
+      return a.localeCompare(b);
+    })
+    .join(',');
+}
+
+/**
+ *
+ * @param {Octokit.ReposCompareCommitsResponseCommitsItem} commitsItem
+ */
+function filterCommit(commitsItem) {
+  // exclude dependabot commits (sometimes they're merged manually so filtering by author is insufficient)
+  return (
+    commitsItem.author.login !== 'dependabot-preview[bot]' &&
+    !commitsItem.commit.message.startsWith('Bump')
+  );
+}
+
+async function main(argv) {
+  const { githubToken, lastRelease, release } = argv;
+
+  if (!githubToken) {
+    throw new TypeError(
+      'Unable to authenticate. Make sure you either call the script with `--githubToken $token` or set `process.eng.GITHUB_TOKEN`.',
+    );
+  }
+  const octokit = new Octokit({
+    auth: githubToken,
+  });
+
+  /**
+   * @type {AsyncIterableIterator<Octokit.Response<Octokit.ReposCompareCommitsResponse>>}
+   */
+  const timeline = octokit.paginate.iterator(
+    octokit.repos.compareCommits.endpoint.merge({
+      owner: 'mui-org',
+      repo: 'material-ui',
+      base: lastRelease,
+      head: release,
+    }),
+  );
+
+  /**
+   * @type {Octokit.ReposCompareCommitsResponseCommitsItem[]}
+   */
+  const commitsItems = [];
+  for await (const response of timeline) {
+    const { data: compareCommits } = response;
+    commitsItems.push(...compareCommits.commits.filter(filterCommit));
+  }
+
+  const authors = Array.from(
+    new Set(
+      commitsItems.map((commitsItem) => {
+        return commitsItem.author.login;
+      }),
+    ),
+  );
+  const contributorHandles = authors.map((author) => `@${author}`).join(', ');
+
+  // We don't know when a particular commit was made from the API.
+  // Only that the commits are ordered by date
+  const commitsItemsByDateAsc = commitsItems.slice();
```

Comment:
nit: I'd expect the Desc order, not sure why :))

Reply:
done